### PR TITLE
[VarExporter] add docs about using proxies in production

### PR DESCRIPTION
**Pull request by kbond**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | n/a

Adds a blurb about the best process for using proxies in production.

/cc @nicolas-grekas (I do not know if this is the most efficient method)

### CODE REVIEW

File: src/Symfony/Component/VarExporter/README.md [hunk 1]
```diff
@@ -139,6 +139,47 @@ $foo = FooLazyProxy::createLazyProxy(initializer: function (): Foo {
...
+    // 2b: return the class
+    return FooProxy::class;
+}
+
+// 3. generate proxy
+// 3a. generate the code
+$proxyCode = FooLazyProxy::generateX(...);
+
+// 3b. write code to file
+\file_put_contents($pathToGeneratedFile, 'class FooLazyProxy'.$proxyCode);
```
Comment (stof) [inline src/Symfony/Component/VarExporter/README.md]: This should be `class FooProxy` to be consistent with other parts of the example

File: src/Symfony/Component/VarExporter/README.md [hunk 2]
```diff
@@ -139,6 +139,47 @@ $foo = FooLazyProxy::createLazyProxy(initializer: function (): Foo {
...
+    // 2a: require the generated file
+    require_once $pathToGeneratedFile;
+
+    // 2b: return the class
+    return FooProxy::class;
+}
+
+// 3. generate proxy
+// 3a. generate the code
+$proxyCode = FooLazyProxy::generateX(...);
```
Comment (stof) [inline src/Symfony/Component/VarExporter/README.md]: shouldn't this be `ProxyHelper` here ?

File: src/Symfony/Component/VarExporter/README.md [hunk 3]
```diff
@@ -139,6 +139,47 @@ $foo = FooLazyProxy::createLazyProxy(initializer: function (): Foo {
...
+    // 2b: return the class
+    return \FooProxy::class;
+}
+
+// 3. generate proxy
+// 3a. generate the code
+$proxyCode = ProxyHelper::generateX(...);
+
+// 3b. write code to file
+file_put_contents($pathToGeneratedFile, 'class FooProxy'.$proxyCode);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarExporter/README.md:168]: missing `<?php` in the generated file

**Review by nicolas-grekas — COMMENTED** [review]
This makes a lot of assumptions. I don't know if this is going to be useful to readers.
Maybe a more extensive article in the doc could be useful?
This is also one way to do it, among many others (DI + Doctrine use their own way as you know)

**Comment by kbond** [comment]
Ok, yeah, adding to https://symfony.com/doc/current/components/var_exporter.html would probably be best.

What I think is missing is: 
1. When using generated proxy code in my app/package, what are my options to optimize for production?
2. What are the pros/cons of each method?
3. What's easiest?

I have several places I want to use proxies but these are the questions I'm stuck on.

Would you consider _this_ option efficient?

**Comment by nicolas-grekas** [comment]
I'd need more context to answer to your questions. I guess the difficult part is figuring out *when* the proxies should be generated: is it during some scaffolding? container compilation? cache warmer? composer install? etc. Eg RedisProxy in Cache is generated once in a while and committed. DI generates during its compilation steps, Doctrine has a command + cache warmer, etc.
It depends on the use case. Closing as this is not mergeable :)

**Pull request closed without merging**